flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporterFactory.java

```diff
@@ -30,11 +30,13 @@
 public class DatadogHttpReporterFactory implements MetricReporterFactory {
 
     private static final Logger LOG = LoggerFactory.getLogger(DatadogHttpReporterFactory.class);
+    private static final String DEFAULT_DATACENTER_DOMAIN = "https://app.datadoghq.%s";
```

Comment:
Perhaps we could make all HTTP-related data live on `DatadogHttpClient`? This way we centralize HTTP knowledge in a single place.